### DISCUSSION TITLE
Does btCompoundShape support removal of child shapes?

### DISCUSSION THREAD

**Discussion by joseph-lys** [bulletphysics/bullet3#4192] [Q&A]
Hi, I am working on a codebase that uses btCompoundShape and removes child shape at some point in time. I was thinking of enabling the AABB tree to improve the performance, however I saw this comment in the header.
https://github.com/bulletphysics/bullet3/blob/5ae9a15ecac7bc7e71f1ec1b544a55135d7d7e32/src/BulletCollision/CollisionShapes/btCompoundShape.h#L54
However, I also see there has been some work on removeChildShape() to get the function to work with AABB tree.

Could some one kindly advise, does the btCompoundShape actually support removal of child shapes?

**Comment by erwincoumans** [comment]
Probably not implemented. Either re-create a new btCompoundShape if needed, or fix the code.

**Reply by joseph-lys** [reply]
Hi @erwincoumans,

Thank you for the reply.

From what I see the that the call to removeChildShape will trigger m_dynamicAabbTree->remove, and the child is popped from m_children. I am not sure what needs to be added, could you kindly point out which part is not implemented/missing in order to properly support removal of child shape?
